From f1c94575317ffac798d4f267c68e3e8f12df10ad Mon Sep 17 00:00:00 2001
From: Aelin <aelin@postmarketos.org>
Date: Sat, 12 Sep 2026 15:31:06 +0200
Subject: [PATCH 10/10] ssh: allow sessions to write systemd inhibit pipes

Allow sshd_session_t to write systemd logind inhibit pipes when
pam_systemd is configured to create sleep inhibitors.
---
 policy/modules/services/ssh.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/services/ssh.te b/policy/modules/services/ssh.te
index 2ec2da3d9..c9534aea3 100644
--- a/policy/modules/services/ssh.te
+++ b/policy/modules/services/ssh.te
@@ -169,6 +169,10 @@ optional_policy(`
 	userdom_map_tmp_files(sshd_session_t)
 ')
 
+optional_policy(`
+	systemd_write_inhibit_pipes(sshd_session_t)
+')
+
 #type sshd_session_devpts_t;
 term_login_pty(sshd_session_devpts_t)
 allow sshd_session_t sshd_session_devpts_t:chr_file { rw_chr_file_perms setattr_chr_file_perms getattr_chr_file_perms relabelfrom };
-- 
2.55.0

